packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -253,6 +253,37 @@ class FileCollection extends DocumentCollection {
     return resp.links.related
   }
 
+  /**
+   * Checks if the file belongs to the parent's hierarchy.
+   *
+   * @param  {string}  childFileID  The file ID of the child
+   * @param  {string}  childDirID   The dirID of the child
+   * @param  {string}  childPath    The child path
+   * @param  {string}  parentID     The parent's document id
+   * @return {boolean}              Whether the file is a parent's child
+   */
+  async isChild(childFileID, childDirID, childPath, parentID) {
```

Comment:
4 parameters, whose order could not be anticipated without the documentation. We may find simething better.

I would say `isChildOf(childId|doc, parentID|doc)` 
- if one gives a doc, then you have the path and the id
- if one gives only the id, then you can send a `stat` request to have the path

Reply:
Ok with your proposition, but I'm a bit surprised by `not be anticipated without the documentation`: how can one use a function without knowing its signature, known by the code or the doc?

Reply:
The rules is that once you know the method, everything should be pretty obvious to read or to write. When you have more than one optionnal parameter, or too many mandatory ones, or when you can't really tell which parameter is what without looking at the doc, you probably should go to named ones (which in JS is by object destructuring)

In our case, the name of the method does not make obvious what and where each parameter  should be others than `child` and `parent`